File: backend/src/SyncChat.Application/Models/Conversation.cs
using System;

namespace SyncChat.Application.Models;

public class Conversation
{
    public string Id { get; set; } = string.Empty;
    public string[] ParticipantUids { get; set; } = Array.Empty<string>();
    public LastMessageInfo? LastMessage { get; set; }
    public DateTime UpdatedAt { get; set; }
}

public class LastMessageInfo
{
    public string Text { get; set; } = string.Empty;
    public string SenderId { get; set; } = string.Empty;
    public DateTime Timestamp { get; set; }
}

File: frontend/app/src/main/java/com/syncchat/app/data/model/Conversation.kt
package com.syncchat.app.data.model

import java.util.Date

data class Conversation(
    val id: String = "",
    val participantUids: List<String> = emptyList(),
    val lastMessage: LastMessageInfo? = null,
    val updatedAt: Date = Date(),
    val isPinned: Boolean = false,
    val isBlocked: Boolean = false,
    val isBlockedByOther: Boolean = false
)

data class LastMessageInfo(
    val text: String = "",
    val senderId: String = "",
    val timestamp: Date = Date()
)
